File: app/src/main/java/iaruchkin/courseapp/ui/NewsDetailsFragment.java
package iaruchkin.courseapp.ui;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.WebView;

import iaruchkin.courseapp.R;
import io.reactivex.disposables.CompositeDisposable;

public class NewsDetailsFragment extends Fragment {
    static final String EXTRA_ITEM_URL = "extra:itemURL";
    private CompositeDisposable compositeDisposable = new CompositeDisposable();
    public static final String TAG = NewsDetailsFragment.class.getSimpleName();
    private MessageFragmentListener listener;

    WebView mWebView;


    public static NewsDetailsFragment newInstance(String itemURL){
        NewsDetailsFragment fragmentFullNews = new NewsDetailsFragment();
        Bundle bundle = new Bundle();
        bundle.putSerializable(EXTRA_ITEM_URL, itemURL);
        fragmentFullNews.setArguments(bundle);
        return fragmentFullNews;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.activity_news_details_webview, container, false);
        mWebView = view.findViewById(R.id.web_view);
        setView(getArguments().getString(EXTRA_ITEM_URL));

        return view;
    }

    public void onStop() {
        super.onStop();
        compositeDisposable.clear();
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof MessageFragmentListener){
            listener = (MessageFragmentListener) context;
        }
    }

    @Override
    public void onDetach() {
        listener = null;
        super.onDetach();
    }

    private void setView(String itemURL){
        if (itemURL != null) {
        mWebView.loadUrl(itemURL);
        }
    }

    private void handleError(Throwable th) {
        Log.e(TAG, th.getMessage(), th);
    }
}

File: app/src/main/java/iaruchkin/courseapp/ui/NewsDetailsFragment.kt
package iaruchkin.courseapp.ui

import android.content.Context
import android.support.v4.app.Fragment
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.webkit.WebView

import iaruchkin.courseapp.R
import io.reactivex.disposables.CompositeDisposable

class NewsDetailsFragment : Fragment() {
    private val compositeDisposable = CompositeDisposable()
    private var listener: MessageFragmentListener? = null

    internal var mWebView: WebView? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.activity_news_details_webview, container, false)
        mWebView = view.findViewById(R.id.web_view)
        setView(arguments!!.getString(EXTRA_ITEM_URL))

        return view
    }

    override fun onStop() {
        super.onStop()
        compositeDisposable.clear()
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        if (context is MessageFragmentListener) {
            listener = context
        }
    }

    override fun onDetach() {
        listener = null
        super.onDetach()
    }

    private fun setView(itemURL: String?) {
        if (itemURL != null) {
            mWebView?.loadUrl(itemURL)
        }
    }

    private fun handleError(th: Throwable) {
        Log.e(TAG, th.message, th)
    }

    companion object {
        internal val EXTRA_ITEM_URL = "extra:itemURL"
        val TAG = NewsDetailsFragment::class.java.simpleName


        fun newInstance(itemURL: String): NewsDetailsFragment {
            val fragmentFullNews = NewsDetailsFragment()
            val bundle = Bundle()
            bundle.putSerializable(EXTRA_ITEM_URL, itemURL)
            fragmentFullNews.arguments = bundle
            return fragmentFullNews
        }
    }
}
